In [ ]:
from IPython.display import Image, display
def display_image(image_path, width):
    display(Image(filename=image_path, width=width))
display_image('path_to_your_image.jpg', width=300)

# Home exercise 1.1. (5p)

Consider the plane stress state below

$$
\sigma = \begin{bmatrix}
\sigma_0 & 0 & 0 \\
0 & \alpha\sigma_0 & 0 \\
0 & 0 & 0
\end{bmatrix}
$$

where $\alpha \in [-1, 1]$, $\sigma_0 > 0$, and $x_1,x_2,x_3$ are the cartesian coordinates in the basis $\vec{e}_1, \vec{e}_2, \vec{e}_3$.

An experimental setting giving rise to such a bi-axial stress state is often used to determine the parameters of material models.

Determine as a function of the parameter $\alpha$

1. The mean stress $\sigma_m$, also called the hydrostatic pressure $p$;
2. The equivalent von Mises stress $\sigma_e = \sqrt{3J_2}$;
3. Lode angle (see Figure 1)

$$
\theta = \frac{1}{3} \arccos\left(\frac{3\sqrt{3}J_3}{2J_2^{3/2}}\right)
$$

Note: pay attention to the definition of the Lode angle (angle between what and what).

4. The maximum shear stress $\tau_{max}$;
5. The normal of the maximum shear stress plane.

What is the stress state corresponding to $\alpha = -1$?


---

In [3]:
import sympy as sp
import numpy as np

In [4]:
# Define the symbols
sigma_0, alpha = sp.symbols('sigma_0 alpha', real=True, positive=True)

# Stress matrix
sigma = sp.Matrix([[sigma_0, 0, 0], 
                   [0, alpha*sigma_0, 0], 
                   [0, 0, 0]])

In [8]:
# 1. Mean stress sigma_m (hydrostatic pressure)
sigma_m = sp.trace(sigma)/3
print(sigma_m)

alpha*sigma_0/3 + sigma_0/3


1. The mean stress $\sigma_m$ as a function of $\alpha$ is

$
\sigma_m = \dfrac{\alpha \sigma_0}{3} + \dfrac{\sigma_0}{3}
$

---

In [9]:
# For the second invariant J2, we only consider the deviatoric components in plane stress
# J2 = (1/2)*((sigma_xx - sigma_yy)**2 + (sigma_yy - sigma_zz)**2 + (sigma_zz - sigma_xx)**2 + 6*(tau_xy**2 + tau_yz**2 + tau_zx**2))

# But since tau_xy = tau_yz = tau_zx = 0 in plane stress, we simplify this to
J2 = (1/6)*((sigma[0,0] - sigma[1,1])**2 + sigma[0,0]**2 + sigma[1,1]**2)

# 2. Equivalent (von Mises) stress sigma_e (simplified for plane stress)
sigma_e = sp.sqrt(3*J2)

print(sigma_e)

0.707106781186548*sqrt(alpha**2*sigma_0**2 + sigma_0**2 + (-alpha*sigma_0 + sigma_0)**2)


2. The equivalent von Mises stress $\sigma_e = \sqrt{3J_2}$ as a function of $\alpha$ is

$\sigma_e = \sqrt{3 \cdot \frac{1}{6} \left(\sigma_{xx} - \sigma_{yy}\right)^2 + \sigma_{xx}^2 + \sigma_{yy}^2}$

0.707106781186548*sqrt(alpha**2*sigma_0**2 + sigma_0**2 + (-alpha*sigma_0 + sigma_0)**2)

---

3. Lode angle

---

4. The maximum shear stress $\tau_{max}$;

---

5. The normal of the maximum shear stress plane.

---

What is the stress state corresponding to $\alpha = -1$?